core/class_db.cpp

```diff
@@ -1115,11 +1109,39 @@ MethodBind *ClassDB::bind_methodfi(uint32_t p_flags, MethodBind *p_bind, const c
 		ERR_EXPLAIN("Method already bound: " + instance_type + "::" + mdname);
 		ERR_FAIL_V(NULL);
 	}
+
 #ifdef DEBUG_METHODS_ENABLED
-	p_bind->set_argument_names(method_name.args);
-	p_bind->set_return_type(rettype);
+	Vector<StringName> arg_type_names = p_bind->get_argument_names();
```

Comment:
Yes, it's filled in the `create_method_bind` which is called before this method.